src/systems/userinput/devices/windows-mixed-reality-controller.js

```diff
@@ -0,0 +1,62 @@
+import { paths } from "../paths";
+import { Pose } from "../pose";
+
+export class WindowsMixedRealityControllerDevice {
+  constructor(gamepad) {
+    this.rayObject = null;
+    this.rayObjectRotation = new THREE.Quaternion();
+
+    // wake the gamepad api up. otherwise it does not report touch controllers.
+    // in chrome it still won't unless you enter vr.
+    navigator.getVRDisplays();
+
+    this.buttonMap = [
+      { name: "touchpad", buttonId: 0 },
+      { name: "trigger", buttonId: 1 },
+      { name: "grip", buttonId: 2 },
+      { name: "menu", buttonId: 3 }
+    ];
+
+    this.gamepad = gamepad;
+    this.pose = new Pose();
+    this.axisMap = [
+      { name: "padX", axisId: 0 },
```

Comment:
I was going to suggest you call these `touchpadX` and `joystickX` for consistency, but looking at our other devices it seems like we did it both ways... I prefer using the same name as the corresponding button, but since we already have inconsistency, its your choice.